src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
+
+    this.logger.debug(
+      `Request Method:[${req.method}] || Request URI:${
+        req.baseUrl
+      } || Request Body:[${JSON.stringify(
+        req.body,
+      )}] || Request Query Params:[${JSON.stringify(req.query)}] Request IP:[${
+        req.ip
+      }] || User Details: [${JSON.stringify(userDetail)}]`,
+    );
```

Comment:
PrimathonAICodeReviewer

The logging of sensitive information like request body, query params, and user details should be done cautiously. Ensure that sensitive data is not logged in clear text for security reasons.
```diff
+     this.logger.debug(
+       `Request Method:[${req.method}] || Request URI:${
+         req.baseUrl
+       } || Request IP:[${
+         req.ip
+       }]`,
+     );
```



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->